### PR TITLE
raftstore: manually compact region after delete a lot of keys.

### PR DESCRIPTION
**Pull request by disksing**
For issue https://github.com/pingcap/tikv/issues/1179

Note: 
This PR does not count deleted keys when GC executes,  but count it when the raft command applies.
A counter is managed for each Peer and it is checked periodically by RaftStore.

@zhangjinpeng1987 @siddontang @BusyJay

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
Please add test.

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -37,55 +37,55 @@ fn main() {
         .author("PingCAP")
         .about("Distributed transactional key value database powered by Rust and Raft")
         .arg(Arg::with_name("db")
-             .short("d")
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: seem you should merge master first.

**Comment by siddontang** [comment]
@zhangjinpeng1987 
I think we should consider compacting raft cf too.

**Comment by zhangjinpeng1987** [comment]
`I think we should consider compacting raft cf too.`
I don't think so, because the raft cf is used frequently, what we need to do is to keep fresh entries in write buffer, but compact range will flush memtable to sst file.

**Comment by siddontang** [comment]
LGTM
PTAL @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**